### PR TITLE
measure distance

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
Adding a possibility to measure distances in the map

### CODE REVIEW

**Comment by taulinger** [comment]
Added the initial infrastructure. The starting point for the feature development is `OlMeasurementHandler.js.

**Comment by taulinger** [comment]
Manually adding the olLayer within the `OlMeasurementHandler` is no longer necessary

File: src/modules/map/services/OlMeasurementHandler.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	 * @returns {VectorLayer} the olLayer which should be attached to
+	 */
+	// eslint-disable-next-line no-unused-vars
+	activate(olMap) {
+		const prepareInteraction = (map) => {
+			const source = new VectorSource({ wrapX: false });	
+			const layer = new VectorLayer({
+				source: source,
+				style: measureStyleFunction,
+				map:map // Sets the layer as overlay on a map
```
Comment (taulinger) [inline src/modules/map/services/OlMeasurementHandler.js:107]: Please remove adding the layer to the map.
With the latest commits, the layer is added by the layer managment in the background

File: src/modules/map/services/OlMeasurementHandler.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+			});
+			return layer;
+		};
+		//use the map to register event listener, interactions, etc
+		//for development purposes you can attach the layer to the map here,
+		//later, this will be done outside this handler
```
Comment (taulinger) [inline src/modules/map/services/OlMeasurementHandler.js:113]: Can be removed now...

File: src/modules/map/services/OlMeasurementHandler.js (hunk 1 above)
Comment (thiloSchlemmer) [reply src/modules/map/services/OlMeasurementHandler.js:107]: if this line is removed, the drawed line will be disappear. The feature is in the layer, but is not rendered.

File: src/modules/map/services/OlMeasurementHandler.js (hunk 2 above)
Comment (thiloSchlemmer) [reply src/modules/map/services/OlMeasurementHandler.js:113]: sure

**Comment by taulinger** [comment]
Well done!
 -  Code structure
 -  Test and test coverage
 - Preparation for later refactoring (common function)


On the one hand the concept of passing the map instance to the handler breaks encapsulation, but on the other hand,
we are able to to do all the necessary and often complicated setup to interact with the map. I think we'll leave that, and I will extract a generic handler interface for other handler developments.

Next steps from my point of view:
- make the tool look and feel the same like the ba3 measurement tool
- add a button in the show case to clear/remove the layer

**Comment by taulinger** [comment]
`OlMeasurementHandler.js` could now extent `OlLayerHandler.js`

**Comment by thiloSchlemmer** [comment]
`OlMeasurementHandler.js` has now Partition-Overlays.

Next Increment is planned with overlay-styling, as for now, the styling is absent. 
Or is it better nor possible to litify the overlay-content to a component, so styling could be moved to this component?

**Review by taulinger — COMMENTED** [review]
We're on track!

**Comment by thiloSchlemmer** [comment]
OlMeasurementHandler have now styling on the overlay-elements and is lifted up to a module.

Next increment is planned with the display of the azimuth-angle in the tooltip

**Comment by taulinger** [comment]
Very nice!!
I just would like to suggest to give the `BaOverlay` class a more meaningful name (e.g. MeasurementOverlay),
for two reasons:

1. We probably will have a couple of "BaOverlay" classes distributed over a couple of folders in the future otherwise
2. You've got a better IDE support when you search for a file, and you just know it has something to do with "measurement"

File: src/modules/map/components/olMap/index.js [hunk 3]
```diff
@@ -1,4 +1,8 @@
 import { OlMap } from './OlMap';
+import { BaOverlay } from './handler/measure/BaOverlay';
 if (!window.customElements.get(OlMap.tag)) {
 	window.customElements.define(OlMap.tag, OlMap);
+}
+if (!window.customElements.get(BaOverlay.tag)) {
+	window.customElements.define(BaOverlay.tag, BaOverlay);
```
Comment (taulinger) [inline src/modules/map/components/olMap/index.js:7]: Can we do the registration also within the handler? A handler should be like a plug-and-play mechanism via DI without anny other dependencies.

Comment (thiloSchlemmer) [reply src/modules/map/components/olMap/index.js:7]: Sure

File: src/modules/map/components/olMap/handler/measure/StyleUtils.js [hunk 4]
```diff
@@ -63,7 +71,8 @@ export const generateSketchStyleFunction = (styleFunction) => {
 			styles = [sketchPolygon];
 		}
 		else if (feature.getGeometry().getType() === 'Point') {
-			const fillColor = [255, 255, 255]; // todo: review, topic for diskussion
+			const globalColor = getComputedStyle(document.body).getPropertyValue('--color');
+			const fillColor = globalColor ? hexToRgb(globalColor) : [0, 0, 0]; // todo: review, topic for diskussion
```
Comment (thiloSchlemmer) [inline src/modules/map/components/olMap/handler/measure/StyleUtils.js:75]: must be discussed: in BA3 the fillcolor was queried from $scope.option

**Review by taulinger — COMMENTED** [review]
Excellent !
Just one thing to note: When the handler is deactivated, the overlays (labels) seems not to be removed completely

Just for me to remind, pending are:
- transformation of geometries to a geodetic SRID for correct length and area calculations
- small refactoring of the LayerHandler (provide an id)

Something more?

**Comment by thiloSchlemmer** [comment]
> Excellent !
> Just one thing to note: When the handler is deactivated, the overlays (labels) seems not to be removed completely
> 
I fixed that

> Just for me to remind, pending are:
> 
> * transformation of geometries to a geodetic SRID for correct length and area calculations
> * small refactoring of the LayerHandler (provide an id)
> 
> Something more?
Nothing from my side, except an possible enhancement for the sketch-styles, if vectorLayers with custom-themes comes to production. Then we could think about retrieving colors from CSS-Globals or stuff like that.

**Comment by thiloSchlemmer** [comment]
Ready for refactoring of the LayerHandler (provide an id)

**Review by taulinger — APPROVED** [review]
Well structured and clean code, very well done! 🥳💪

**Pull request merged**